### ISSUE TITLE
Marshmallow's default values are not changed by Freezegun

### ISSUE THREAD

**Issue by egor83** [spulec/freezegun#372]
Minimal reproducible example:

app_stub/__init__.py:
https://pastebin.com/TYbyBhwL

app_stub/serializers.py:
https://pastebin.com/vNS9aAuj

tests/conftest.py:
https://pastebin.com/kqia2pKH

tests/test_stub.py
https://pastebin.com/SVxty54j

Or I could send you a zip file if you give me an email address.

I would expect Marshmallow's default values to be set to Freezegun's time, but it uses real time instead.
At a glance, it seems to happen because Marshmallow's default value is (evaluated? loaded?) during test setup, and not during test execution, so probably before Freezegun comes into play - you can check this if you uncomment exception throwing in Marshmallow's default function.

That said - thanks a lot for Freezegun, it's been a great help for us!

**Comment by boxed** [comment]
At work we handle situations like that by not evaluating values like this at import time if at all possible. If it's not possible you can do another trick which is to freeze the time by default in tests and do it super early so other libraries aren't imported until after. 

Here's an article I wrote on this: https://kodare.net/2018/09/12/flaky-tests-part-3-freeze-the-world.html

**Comment by egor83** [comment]
Thanks for response!

> another trick which is to freeze the time by default in tests and do it super early so other libraries aren't imported until after

Can you please elaborate? I added the following:
```
from freezegun import freeze_time
from datetime import datetime
freezer = freeze_time("2020-09-13T13:00:00.000")
print("app", datetime.now())
1/0
```
to my `app.py` file before all imports, but it still prints current time, not frozen one.

Furthermore, Freezegun's doc says:
https://github.com/spulec/freezegun#default-arguments
> Note that FreezeGun will not modify default arguments. The following code will print the current date.

So I'm guessing it's not at all possible to change default values, or do I misunderstand something?

> we handle situations like that by not evaluating values like this at import time if at all possible

At the moment, I am just adjusting our tests to always pass the time explicitly, avoiding the default values ever being used in tests.

**Comment by egor83** [comment]
> `E TypeError: type 'pandas._libs.tslibs.base.ABCTimestamp' is not dynamically allocated but its base type 'FakeDatetime' is dynamically allocated`

I fixed it with a bit descibed [here](https://github.com/spulec/freezegun/issues/98#issuecomment-176006555):
> I got it to work by explicitly importing Pandas before the test code runs.

Thanks for help again!

**Comment by egor83** [comment]
Although now that I think about it, freezing time globally (even for test environment only) might be an overkill for us; but that's another story.

**Comment by boxed** [comment]
Overkill is way better than underkill. Freezing by default means predictable tests by default.

**Comment by egor83** [comment]
Fair enough!

**Comment by egor83** [comment]
I've just checked - further freezes override the first one, so global freeze indeed looks like a good solution (we just already have a ton of tests using Freezegun, so if global freeze would break these, it'd be a hassle - but it's not what happens).

**Comment by boxed** [comment]
Yes, nested freezes is a big part of the API! With a global freeze you can also simplify your tests quite a bit.
